src/nodes/dir.rs

```diff
@@ -119,10 +119,20 @@ impl Node for Dir {
     fn getattr(&self) -> NodeResult<fuse::FileAttr> {
         let mut state = self.state.lock().unwrap();
 
-        let check_type = |t: fs::FileType| if t.is_dir() { Ok(()) } else { Err("directory") };
-        if let Some(attr) = super::get_new_attr(
-            self.inode, state.underlying_path.as_ref(), check_type)? {
-            state.attr = attr;
+        let new_attr = match state.underlying_path.as_ref() {
```

Comment:
This should use `Option.map`. Once NLL is available, you can directly assign instead.

Reply:
Interesting... but can it? Note the error handling within this match pattern. I don't think I can put that within the map lambda?

Reply:
Curses, foiled again.